### ISSUE TITLE
hidden external modules

### ISSUE THREAD

**Issue by DirkdeDraak** [hpcugent/easybuild-framework#1483]
My Intel compiler toolchain uses Hidden Modules as dependencies, also being marked External.

i/icc/icc-2015.2.164-GCC-4.9.2.eb
```
dependencies = [
    ('licenses/.intel', EXTERNAL_MODULE),
]
```

Trying to do a Dryrun of an easyconfig that relies on that toolchains fails.
```
eb q/Qt/Qt-5.4.1-intel-2015.02.eb -Dr
== temporary log file in case of crash /tmp/eb-lraac4/easybuild-I_wio1.log
ERROR: Missing modules for one or more dependencies marked as external modules: ['licenses/.intel']
```

It seems dry-run can't cope with hidden modules just yet?

When starting the build (without Dryrun option) it works fine.

Using EB v2.4.0

**Comment by DirkdeDraak** [comment]
With this patch EB v2.4.0 will correctly recognize hidden external modules:

```
--- easybuild_framework-2.4.0-py2.6.egg/easybuild/framework/easyconfig/easyconfig.py.orig       2015-11-12 13:23:15.431791454 +0100
+++ easybuild_framework-2.4.0-py2.6.egg/easybuild/framework/easyconfig/easyconfig.py    2015-11-27 12:02:00.140065479 +0100
@@ -1176,6 +1176,8 @@
         if ec.get('external_module', False):
             # external modules have the module name readily available, and may lack the info required by the MNS
             mod_name = ec['full_mod_name']
+            if mod_name.split('/')[-1].startswith('.'):
+                ec['hidden']= True
             self.log.debug("Full module name for external module: %s", mod_name)
         else:
             mod_name = self._det_module_name_with(self.mns.det_full_module_name, ec, force_visible=force_visible)
```

**Comment by boegel** [comment]
@DirkdeDraak: thanks for reporting this and suggesting a fix, we'll try and get this in ASAP

**Comment by DirkdeDraak** [comment]
@boegel : I found another place where hidden external modules don't work yet, although that is with the experimental --minimal-toolchains.
Note that I have my own logic for setting the license files via Environment Modules implemented.

I have most components for the intel/2016a toolchain in place (I use intel/2016.01 at the moment), so I want to check if all dependencies are met.
```
eb --experimental --minimal-toolchains i/intel/intel-2016a.eb -Dr
```
gives:
```
ERROR: Missing modules for one or more dependencies marked as external modules: ['licenses/.intel']
```

Last part of the logfile:
```
== 2016-01-22 13:31:42,317 runpy.ActiveMNS DEBUG Determining full module name for {'external_module_metadata': {}, 'full_mod_name': 'licenses/.intel', 'toolchain': None, 'name': None, 'dummy': False, 'short_mod_name': 'licenses/.intel', 'external_module': True, 'versionsuffix': '', 'version': None, 'hidden': False, 'build_only': False} (force_visible: False)
== 2016-01-22 13:31:42,317 runpy.ActiveMNS DEBUG Full module name for external module: licenses/.intel
== 2016-01-22 13:31:42,317 runpy.easyconfig.tools DEBUG Treating dependency marked as external dependency as resolved: {'external_module_metadata': {}, 'full_mod_name': 'licenses/.intel', 'toolchain': None, 'name': None, 'dummy': False, 'short_mod_name': 'licenses/.intel', 'external_module': True, 'versionsuffix': '', 'version': None, 'hidden': True, 'build_only': False}
== 2016-01-22 13:31:42,375 runpy ERROR Missing modules for one or more dependencies marked as external modules: ['licenses/.intel'] (at easybuild/EasyBuildApp/DEVELOPMENT/software/EasyBuild/2.5.0/lib/python2.6/site-packages/easybuild_framework-2.5.0-py2.6.egg/easybuild/tools/robot.py:189 in resolve_dependencies)
```

**Comment by boegel** [comment]
@DirkdeDraak: thanks for reporting, I'll try and make sure both issues are fixed

**Comment by boegel** [comment]
ah, this is a duplicate of https://github.com/hpcugent/easybuild/issues/185

@DirkdeDraak: are you still seeing either issues with the patch from https://github.com/hpcugent/easybuild-framework/pull/1552 included?

**Comment by boegel** [comment]
assuming this is fixed with #1552
